### ISSUE TITLE
Add option to view review using git diff on repository

### ISSUE THREAD

**Issue by ejose19** [Morganamilo/paru#210]
As users may have a custom tool for viewing diffs in git (like ```delta```, ```diffr```, ```diff-so-fancy``` and so son) it is desired to have the same functionability when reviewing the diffs in paru. 

An option that replaces the default command with ```git diff AUR_SEEN..HEAD@{u}```, using git empty tree digest if the repo hasn't been cloned would accomplish this (as suggested by https://github.com/Morganamilo/paru/issues/143#issuecomment-772907935)

**Comment by Morganamilo** [comment]
Does not just setting the diff-viewer program and pager work for this? Diffing from empty branch is what was done before, but it's nicer to print it whole with syntax highlighting.

**Comment by ejose19** [comment]
@Morganamilo good point regarding empty branch, however advanced diff-viewers not only show the green background, but also have syntax highlight included, so having it being an option for empty branch is desirable.

>Does not just setting the diff-viewer program and pager work for this?
I've tried with many combinations in global git config, yet they're either ignored or the arguments sent to ```git diff``` are not as simple as ```AUR_SEEN..HEAD@{u}```, making it fail. Worth noting that calling ```git diff``` directly in the repo shows the correct output.

**Comment by Morganamilo** [comment]
The exact command used is `git diff --stat --patch --cached --color=always -- :!.SRCINFO`.

By any chance did you only set the diff program for interactive mode? Given paru is capturing the output that wouldn't trigger.

**Comment by ejose19** [comment]
an example of my global ```.gitconfig```:

```.gitconfig
[core]
    editor = micro
    pager = delta

[interactive]
    diffFilter = delta --color-only
```

How one would set it for non-interactive as well? I tried setting the ```[diff] external =``` option in git but that errors out (the arguments sent are completely different)

**Comment by Morganamilo** [comment]
Not sure exactly what the config for git is. However, another way may be better running `PAGER="delta --color-only | less` (yes there needs to be a pager setting in parur.conf).

However this hangs infinity and I have no idea why, need to figure out what's wrong.

**Comment by Morganamilo** [comment]
Thinking about it, if delta makes use of PAGER then it could be looping...

**Comment by ejose19** [comment]
Yes, it hangs on my end as well. Also worth noting that calling this command directly ```git diff --stat --patch --cached --color=always -- :!.SRCINFO``` (given the changes are staged) shows me the correct output, where in paru it returns the "simplest" output.

**Comment by Morganamilo** [comment]
`git diff --stat --patch --cached --color=always -- :!.SRCINFO | cat` would give the incorrect outcome though I assume?

Anyway my guess was right. Works if I set it via paru.conf. That will be coming soon.

**Comment by Morganamilo** [comment]
```
[bin]
Pager = delta --color-only | delta
```

This should work in paru.conf now. I'm also considering adding $PARU_PAGER to avoid this issue.

**Comment by ejose19** [comment]
> ```git diff --stat --patch --cached --color=always -- :!.SRCINFO | cat```

Exactly, this is what I see in paru output. Btw why there's a need for "delta | delta"?

**Comment by Morganamilo** [comment]
Going off of the git config `delta --color-only` is used as the diff filter. then you also have `delta` as the pager. So the diff filter is piped into the pager. You're free to configure it however you like.

**Comment by ejose19** [comment]
Sure, will build from master and test if it's necessary to have ```delta | delta``` or the output generated by ```git diff``` (which is picking ```delta --color-only``` in its config) is already enough to only use ```Pager = delta``` in ```paru.conf```

**Comment by ejose19** [comment]
Yes! it works correctly with just 

```
[bin]
Pager = delta
```

Btw, for empty tree the only difference between git diff and paru output is:
- doesn't show all files (already tracked in #209)
- file name is not printed (i guess since it's using ```bat PKGBUILD``` instead of ```find ! -path './.git/*' ! -name .SRCINFO -exec bat {} \;```)

so for empty tree there's not much gain after #209 is solved, so this can be closed. Thanks @Morganamilo!
